src/bam/record.rs

```diff
@@ -321,22 +320,29 @@ impl Record {
         let q_len = qname.len() + 1;
         let extranul = extranul_from_qname(qname);
 
-        self.inner_mut().l_data = (q_len
-            + extranul
-            + cigar_width
-            + ((seq.len() as f32 / 2.0).ceil() as usize)
-            + qual.len()) as i32;
-
-        assert!(qname.len() <= 256);
-
+        let orig_aux_offset = self.qname_capacity()
+            + 4 * self.cigar_len()
+            + (self.seq_len() + 1) / 2
+            + self.seq_len();
+        let new_aux_offset = q_len + extranul + cigar_width + (seq.len() + 1) / 2 + qual.len();
+        let aux_len = self.inner.l_data as usize - orig_aux_offset;
```

Comment:
OK, I do believe that is is _impossible_ for `l_data` to be `< orig_aux_offset`. Subtractions of unsigned like this are always scary--probably best to communicate this in a comment.

Reply:
Agreed. I'll add an assert `assert!(orig_aux_offset <= self.inner.l_data);`